cpp/daal/src/algorithms/dbscan/oneapi/cl_kernels/dbscan_cl_kernels.cl

```diff
@@ -27,161 +27,93 @@
-                                       __global int * assignments, __global int * queue) {
-        const int subgroup_index  = get_global_id(0) * get_num_sub_groups() + get_sub_group_id();
-        const int subgroup_offset = subgroup_index * chunk_size;
-        point_id                  = first_chunk_offset < 0 ? point_id : queue[point_id];
-        const int dist_offset     = subgroup_offset + (first_chunk_offset < 0 ? 0 : first_chunk_offset);
-        if (subgroup_offset >= num_points) return;
-
-        const int subgroup_point_number                     = num_points - subgroup_offset < chunk_size ? num_points - subgroup_offset : chunk_size;
-        const int subgroup_size                             = get_sub_group_size();
-        const int local_id                                  = get_sub_group_local_id();
-        __global const algorithmFPType * subgroup_distances = &distances[dist_offset];
-        __global int * subgroup_assignments                 = &assignments[subgroup_offset];
-        const int subgroup_chunk_offset                     = chunk_offsets[subgroup_index];
-        int local_offset                                    = 0;
-        for (int i = local_id; i < subgroup_point_number; i += subgroup_size)
-        {
-            int is_nbr           = get_nbr_status(subgroup_distances, i, eps);
-            int is_undefined_nbr = get_undefined_nbr_status(subgroup_assignments, point_id, i, is_nbr, subgroup_offset);
-            int local_pos        = sub_group_scan_exclusive_add(is_undefined_nbr);
-            if (is_undefined_nbr)
+            int index           = queue[j];
+            algorithmFPType sum = 0.0;
+            for (int i = localId; i < numFeatures; i += subgroupSize)
             {
-                subgroup_assignments[i]                                                      = cluster_id;
-                queue[queue_last_element + subgroup_chunk_offset + local_offset + local_pos] = subgroup_offset + i;
+                algorithmFPType val = points[subgroupIndex * numFeatures + i] - points[index * numFeatures + i];
+                sum += val * val;
             }
-            if (is_nbr && (subgroup_assignments[i] == _NOISE_))
+            algorithmFPType distance = sub_group_reduce_add(sum);
+            if (distance > eps) continue;
+            if (localId == 0)
             {
-                subgroup_assignments[i] = cluster_id;
+                clusters[subgroupIndex] = clusterId;
             }
-            local_offset += sub_group_reduce_add(is_undefined_nbr);
-        }
-        if (subgroup_index == 0 && local_id == 0)
-        {
-            assignments[point_id] = cluster_id;
+            if (cores[subgroupIndex] == 0) continue;
+            if (localId == 0)
+            {
+                int newIndex    = atomic_inc(counterPtr);
+                queue[newIndex] = subgroupIndex;
+            }
+            break;
```

Comment:
Can we group all code grouped into `if (localId == 0)...`  into one branch?

Reply:
It would result in nested branches